### PR TITLE
Do not apply ExplicitBucketHistogramConfiguration to non-histograms

### PR DESCRIPTION
**Pull request by alanwest**

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by alanwest** [comment]
Oops sorry. Just pushed the fix, so requesting another review.

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 1]
```diff
@@ -162,6 +162,12 @@ internal sealed class MeterProviderSdk : MeterProvider
                             try
                             {
                                 metricStreamConfig = viewConfig(instrument);
+
+                                if (metricStreamConfig is ExplicitBucketHistogramConfiguration
+                                    && instrument.GetType().GetGenericTypeDefinition() != typeof(Histogram<>))
+                                {
+                                    metricStreamConfig = null;
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:169]: Q: Should we log an event for this to let the user know they might have made a mistake?

Comment (cijothomas) [reply src/OpenTelemetry/Metrics/MeterProviderSdk.cs:169]: `OpenTelemetrySdkEventSource.Log.MetricViewIgnored` See! I added this predicting future demand 😆

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Fix issue where `ExplicitBucketHistogramConfiguration` could be used to
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:5]: We need to message the wording that -- as of today, changing aggregations are not supported. This PR is making sure that unsupported Views are ignored, falling back to defaults..
Some wording to indicate that a future version can allow changing aggregations...

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 3]
```diff
@@ -162,6 +162,16 @@ internal sealed class MeterProviderSdk : MeterProvider
...
                             {
                                 metricStreamConfig = viewConfig(instrument);
+
+                                if (metricStreamConfig is ExplicitBucketHistogramConfiguration
+                                    && instrument.GetType().GetGenericTypeDefinition() != typeof(Histogram<>))
+                                {
+                                    OpenTelemetrySdkEventSource.Log.MetricViewIgnored(
+                                        instrument.Name,
+                                        instrument.Meter.Name,
+                                        "Histogram metric stream configuration can not be used for instruments that are not histograms.",
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:172]: "The current SDK does not allow aggregating non-Histogram instruments as Histograms.." - Would this look nicer?

Comment (alanwest) [reply src/OpenTelemetry/Metrics/MeterProviderSdk.cs]: Yes, agreed that's nicer.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry/Metrics/MeterProviderSdk.cs [hunk 4]
```diff
@@ -162,6 +162,16 @@ internal sealed class MeterProviderSdk : MeterProvider
                             try
                             {
                                 metricStreamConfig = viewConfig(instrument);
+
+                                if (metricStreamConfig is ExplicitBucketHistogramConfiguration
+                                    && instrument.GetType().GetGenericTypeDefinition() != typeof(Histogram<>))
+                                {
```
Comment (cijothomas) [inline src/OpenTelemetry/Metrics/MeterProviderSdk.cs:168]: metricStreamConfig = null; -- this statement got lost in new commit.

Comment (alanwest) [reply src/OpenTelemetry/Metrics/MeterProviderSdk.cs:168]: 🤦

**Pull request merged**